python/lib/scheduler/dmod/scheduler/scheduler.py

```diff
@@ -237,6 +268,169 @@ def build_host_list(job: 'Job') -> str:
+            msg = "Attempting to start {} job {} with fewer than allowed minimum of {} required {} datasets."
+            raise RuntimeError(msg.format(job.model_request.__class__.__name__, job.job_id, min_count, category))
+        elif max_count is not None and len(dataset_names) > max_count:
+            msg = "Attempting to start {} job {} with more than allowed max of {} required {} datasets."
+            raise RuntimeError(msg.format(job.model_request.__class__.__name__, job.job_id, max_count, category))
+        elif None in dataset_names is None:
+            msg = "Attempting to start {} job {} with unfulfilled {} data requirement."
+            raise RuntimeError(msg.format(job.model_request.__class__.__name__, job.job_id, category))
+        # If things look good, return the set of names we found after converting to a list
+        else:
+            return list(dataset_names)
+
+    # TODO (later): once we get to dynamic/custom images (i.e., for arbitrary BMI modules), make sure this still works
+    def _generate_docker_cmd_args(self, job: 'Job', worker_index: int) -> List[str]:
+        """
+        Create the Docker "CMD" arguments list to be used to start all services that will perform this job.
+
+        Docker "CMD" arguments are the arguments that will be passed to the Docker entrypoint script/executable when
+        starting a container.  This function essentially generates the appropriate arguments for the applicable
+        entrypoint script, in order to start the worker service at the given index, among the collection of all worker
+        services involved with executing the given job.
+
+        Parameters
+        ----------
+        job : Job
+            The job to have worker Docker services started, with those services needing "CMD" arguments generated.
+        worker_index : int
+            The particular worker service index in question, which will have a specific set of data requirements.
+
+        Returns
+        -------
+        List[str]
+            A list of the Docker "COMMAND" args to be used when creating the worker service at the given index.
+
+        See Also
+        -------
+        https://docs.docker.com/engine/reference/builder/#cmd
+        https://docs.docker.com/engine/reference/builder/#understand-how-cmd-and-entrypoint-interact
+        """
+        # TODO (later): handle non-model-exec jobs in the future
+        if job.model_request.event_type != MessageEventType.MODEL_EXEC_REQUEST:
+            raise RuntimeError("Unsupported requested job event type {}; cannot generate Docker CMD arg values".format(
+                job.model_request.get_message_event_type()))
+
+        # TODO (later): have something more intelligent than class type to determine right entrypoint format and
+        #  values, but for now assume/require a "standard" image
+        if isinstance(job.model_request, NWMRequest) or isinstance(job.model_request, NGENRequest):
```

Comment:
should this be `not isinstance(....) or not isinstance(...)`?

Reply:
Yes, thanks.  Fixed.